lib/MusicBrainz/Server/Data/Editor.pm

```diff
@@ -286,21 +286,25 @@ sub update_email
 {
     my ($self, $editor, $email) = @_;
 
-    Sql::run_in_transaction(sub {
-        if ($email) {
-            my $email_confirmation_date = $self->sql->select_single_value(
-                'UPDATE editor SET email=?, email_confirm_date=NOW()
-                WHERE id=? RETURNING email_confirm_date', $email, $editor->id);
-            $editor->email($email);
-            $editor->email_confirmation_date($email_confirmation_date);
-        }
-        else {
-            $self->sql->do('UPDATE editor SET email=NULL, email_confirm_date=NULL
-                      WHERE id=?', $editor->id);
-            delete $editor->{email};
-            delete $editor->{email_confirmation_date};
-        }
-    }, $self->sql);
+    if ($editor->deleted) {
+        $c->detach('/user/not_found');
+    } else {
+        Sql::run_in_transaction(sub {
```

Comment:
No need to put this in an `else` actually since `$c->detach` always dies.  Would make the diff more readable. :)